tests/queries/0_stateless/02499_extract_key_value_pairs_multiple_input.sql

```diff
@@ -0,0 +1,428 @@
+-- { echoOn }
+
+-- basic tests
+
+-- expected output: {'age':'31','name':'neymar','nationality':'brazil','team':'psg'}
+WITH
+    extractKeyValuePairs('name:neymar, age:31 team:psg,nationality:brazil') AS s_map,
+    CAST(
+            arrayMap(
+                    (x) -> (x, s_map[x]), arraySort(mapKeys(s_map))
+                ),
+            'Map(String,String)'
+        ) AS x
+SELECT
+    x;
+
+-- keys and values starting with number, underscore and other special characters
+-- expected output: {'$nationality':'@brazil','1name':'neymar','4ge':'31','_team':'_psg'}
+WITH
+    extractKeyValuePairs('1name:neymar, 4ge:31 _team:_psg,$nationality:@brazil') AS s_map,
+    CAST(
+            arrayMap(
+                    (x) -> (x, s_map[x]), arraySort(mapKeys(s_map))
+                ),
+            'Map(String,String)'
+        ) AS x
+SELECT
+    x;
+
+-- only special characters
+-- expected output: {'#':'#','$':'$','@':'@','_':'_'}
+WITH
+    extractKeyValuePairs('_:_, @:@ #:#,$:$') AS s_map,
+    CAST(
+            arrayMap(
+                    (x) -> (x, s_map[x]), arraySort(mapKeys(s_map))
+                ),
+            'Map(String,String)'
+        ) AS x
+SELECT
+    x;
+
+-- special (not control) characters in the middle of elements
+-- expected output: {'age':'3!','name':'ney!mar','nationality':'br4z!l','t&am':'@psg'}
+WITH
+    extractKeyValuePairs('name:ney!mar, age:3! t&am:@psg,nationality:br4z!l') AS s_map,
+        CAST(
+            arrayMap(
+                (x) -> (x, s_map[x]), arraySort(mapKeys(s_map))
+            ),
+            'Map(String,String)'
+        ) AS x
+SELECT
+    x;
+
+-- non-standard escape characters (i.e not \n, \r, \t and etc), back-slash should be preserved
+-- expected output: {'amount\\z':'$5\\h','currency':'\\$USD'}
+WITH
+    extractKeyValuePairs('currency:\$USD, amount\z:$5\h') AS s_map,
+    CAST(
+            arrayMap(
+                    (x) -> (x, s_map[x]), arraySort(mapKeys(s_map))
+                ),
+            'Map(String,String)'
+        ) AS x
+SELECT
+    x;
+
+-- invalid escape sequence, should be discarded
+-- expected output: {'valid_key':'valid_value'}
+WITH
+    extractKeyValuePairsWithEscaping('valid_key:valid_value key:invalid_val\\', ':', ' ', '"') AS s_map,
```

Comment:
Why don't we treat `\\` at the end as garbage and not ignore it and parse `key:invalid_val` ?

Reply:
Done